homeassistant/components/weatherkit/weather.py

```diff
@@ -0,0 +1,255 @@
+"""Weather entity for Apple WeatherKit integration."""
+
+from types import MappingProxyType
+from typing import Any
+
+from homeassistant.components.weather import (
+    Forecast,
+    SingleCoordinatorWeatherEntity,
+    WeatherEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LOCATION,
+    CONF_LONGITUDE,
+    CONF_NAME,
+    UnitOfLength,
+    UnitOfPressure,
+    UnitOfSpeed,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import ATTRIBUTION, DOMAIN
+from .coordinator import WeatherKitDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a weather entity from a config_entry."""
+    coordinator: WeatherKitDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+    async_add_entities(
+        [
+            WeatherKitWeather(
+                coordinator,
+                config_entry.data,
```

Comment:
It would be cleaner to put the data in the coordinator so you only have to pass around one object